Fix accesibility issues

AlexVelezLl opened the pull request:
## Summary
* Focus first focusable element on subpages navigation on Lesson and quiz resources selection
* Add `go back` aria label to back buttons
* Add quiz selection change anouncement through aria-live=polite
* Remove d&d widgets in the PreviewSelectedResources page.

## References
Closes #13145

## Reviewer guidance
Check that all points in #13145 have been resolved (but the ones that have other follow-up issues)

For code review:
* I have added an `autofocusFirstEl` directive, based on the [logic we have in KTable to find focusable elements](https://github.com/learningequality/kolibri-design-system/blob/406705c3b4bc3bf9463d369f046c4bcc7cf551cc/lib/KTable/index.vue#L539) to focus the first focusable element in the page when the given node is inserted. This could have also been done with a composable instead of a directive, but I have chosen directives because with this I just needed to define it in the parent component and set the v-autofocus-first-el to the `router-view`, with composables I needed to call the composable in every subpage. How does it look? Is there any argument to implement this in a composable or in any other way?

AlexVelezLl committed:
Focus first focusable element when navigating on side panels

AlexVelezLl committed:
Add sendpolitemessage on quiz selection change

AlexVelezLl committed:
Remove drag widgets from selected resources list

AlexVelezLl committed:
Remove unnecessary code

marcellamaki commented on kolibri/plugins/coach/assets/src/views/common/directives/autofocusFirstEl.js:
```diff
@@ -0,0 +1,49 @@
+  if (el.tabIndex < 0) {
+    return false;
+  }
+  if (el.offsetParent === null && window.getComputedStyle(el).position !== 'fixed') {
```
Why are we checking `fixed` position here?

AlexVelezLl replied:
I have added another line explaining why to skip if the position is fixed :) in the link we can see more info but I was not sure how much should we explain here. Is a little bit more understandable now?

marcellamaki replied:
Yes - that's great, thanks Alex. I think you have a good balance of inline comment details plus the link for more information now :)

marcellamaki reviewed (changes requested):
Hi @AlexVelezLl  - I really like the approach here, and it's nice to try out using a directive in a tightly scoped way :) 

A few notes from my best effort on the manual QA side. 

First, the items that need to be resolved: 

> If the user workflow contains several views within the sidebar (for example while navigating back and forth through the folders), focus order should be logical: when user selects one folder and presses Enter to open and navigate within, the focus in the next step (view) should be placed on top of the sidebar (maybe the Search button), and not at the bottom (the Save & finish button).

It seems like generally speaking this is fixed 👍 nice job. I just see two small issues. 

1. **Not necessarily blocking, but some investigation would be good:** it seems like the "go back" arrow is always out of the focus trap, and like it's somehow it it's own focus trap at the top. When I "loop through" the tab order within the modal, I never hit the back arrow. To get to it, I need to "shift + Tab" from the "X" button. I _think_ we want to include it in the focus trap. This seems to be a universal problem across the side panel modal - maybe it has to do with our use of slots. This doesn't necessarily have to be blocking merge, and we could perhaps go ahead and have Radina weigh in when she returns. 

https://github.com/user-attachments/assets/3c1ff41c-0004-47bc-92be-82b963d7547b

2. **Should fix:** I think there's some small bug in the autofocus. My best guess is that maybe it's colliding in some places with the previously existing FocusTrap logic in the side panel modal, i.e. 

```
<KFocusTrap
        @shouldFocusFirstEl="focusFirstEl"
        @shouldFocusLastEl="focusLastEl"
      >
```

My very preliminary guess is that some combination of this existing logic, the autofocus, and the nextTick/order of execution is causing some "automatic" shifting, as well as it disappearing sometimes. I'd be happy to dig into it a bit further with you and talk through what we might want to do going forward - I don't really have a clear idea yet if we should be removing some of that pre-existing code in favor of the directive that you implemented, or working to align the directive more seamlessly with the existing code... (Also possible that isn't the issue and I just haven't explored deeply enough)

https://github.com/user-attachments/assets/bef15290-5986-4586-922e-a160c5d767d6


> When one or more resources are selected, the link that appears at the bottom of the sidebar (N resources selected (...KB)) must be fully announced through the aria-live="polite" region, including the size of the resources (as of now it only state the number of selected resources).

Overall, the code change seems appropriate and I tried this with VoiceOver (apples default screenreader) and the aria polite interruption was working as expected. 
**String/aria update:** it still doesn't have the size of the resources in lessons. We don't worry about this in quizzes, (presumably because we're less concerned about file size). On the lessons we do want to make sure to use the string that includes the file size, because the total size of the lesson being synced is very relevant, and some resources could be quite heavy. 

------

Things that all seem fixed! 🎉 

> Top left back button (left arrow icon) missing now has a label but it should be changed to [Go back](https://github.com/learningequality/kolibri/blob/a7066b85e361329b29c7cbd0eaff816cee632584/packages/kolibri/uiText/commonCoreStrings.js#L81) instead of just Back.

looks good ✅ 

> maybe we can skip the feature to reorder resources in this page, since the user still has the option to do it on the lesson summary page...?  
✅ 

> If the user opens the resource preview from the above "selected resource review" step, there is no back button anymore in front of the Manage lesson resources heading. We should add it for consistency, and also because even though those who use the mouse can still easily click on the N resources selected (...KB) link at the bottom of the sidebar, keyboard navigation users would benefit from having a closer button to go back to the previous step.

Looks good ✅

AlexVelezLl committed:
Disable autofocus on first landing route

AlexVelezLl committed:
Announce selection size on lesson resource selection

AlexVelezLl committed:
Add explanation for offsetParent null

AlexVelezLl commented:
Hi @marcellamaki!

Point 1 will need some more considerations since the reason of that bug is because inside the side panel itself we literaly set the close button as first and last element to focus, and changing this will require us to take a look at the implications on all places where we use side panels. I can do that in the scope of this PR if that makes sense, or I can take a look in a follow up if we dont want this to be a blocker for now.

https://github.com/AlexVelezLl/kolibri/blob/766b90f57fe652f03b01a4cfa3039bf1d4650f91/packages/kolibri-common/components/SidePanelModal/index.vue#L208-L217

For point 2 I have fixed it by avoiding the autofocus if the current page is the "landing" route, i.e. is the first route we are seeing after opening the side panel https://github.com/learningequality/kolibri/pull/13246/commits/8c36a23b1754bd04db86ab242f41cd3172e77fd3. This makes that the first time we open the side panel, the focus will always land on the close button without any race condition of who takes the focus first, but in all following navigations the first focusable elements of the subpages are the ones that will take the focus.

I have also added the size of selected resources in the polite message https://github.com/learningequality/kolibri/pull/13246/commits/44e071542989fc2e090ea948e388190d1f362ef2, didnt know that we were missing this in the message :sweat_smile:.

marcellamaki commented:
That sounds good Alex - thank you! I think for point 1, let's not have this be blocking, both because of the technical complexity and potential for scope creep, but also because I do want to ask @radinamatic when she's feeling better about her insight into this. It's possible that being able to use the back tab approach is okay for now, and we can think about how this might fit in with some of the planned "polishing up"/refactoring work that we've discussed previously with @nucleogenesis and others for the planned patch releases. So, I think doing this in follow up will be fine, and we'll just have to decide how urgent it is. 

I'll rereview the last few points shortly. Thanks! :)

marcellamaki reviewed (approved):
Thank you Alex! confirmed the two updates discussed are fixed! We'll handle the back button separately and discuss with Radina.
